src/univariate/continuous/normal.jl

```diff
@@ -1,12 +1,18 @@
-immutable Normal <: ContinuousUnivariateDistribution
-    μ::Float64
-    σ::Float64
+immutable Normal{T<:Number} <: ContinuousUnivariateDistribution
+    μ::T
+    σ::T
 
-    Normal(μ::Real, σ::Real) = (@check_args(Normal, σ > zero(σ)); new(μ, σ))
-    Normal(μ::Real) = Normal(μ, 1.0)
-    Normal() = Normal(0.0, 1.0)
+    Normal(μ::T, σ::T) = new(μ, σ)
 end
 
+Normal{T<:Number}(μ::T, σ::T) = Normal{T}(μ, σ)
+Normal(μ::Number, σ::Number) = Normal(promote(μ, σ)...)
+Normal(μ::Number) = Normal(μ, one(μ))
+#Normal(μ::Real, σ::Real) = (@check_args(Normal, σ > zero(σ)); Normal(μ, σ))
+Normal(μ::Real) = Normal(μ, one(μ))
+Normal() = Normal(0.0, 1.0)
+Normal(μ::Complex, σ::Complex) = error("Normal with complex mean and variance not defined")
```

Comment:
> So is it kosher to say that dual numbers are a subtype of Real?

I don't think we can say `Dual <: Real`; dual numbers aren't a subset of the reals, they're ([a very special](http://math.stackexchange.com/questions/1196775/having-trouble-understanding-generalized-complex-numbers)) subset of the generalized complex numbers. 

However, I'm really growing on the idea that we should define `ForwardDiffNumber <: Real`. The purpose of  ForwardDiff is to do AD, not make algebraically correct statements about dual numbers - that should be the purview of the DualNumbers package. From a practical perspective, we already just drop in `ForwardDiffNumber`s in place of `Real`s - so, computationally, `ForwardDiffNumber` should lie under `Real` in the type hierarchy, even though these numbers are not algebraically a subset of the real numbers.

My main issue with `ForwardDiffNumber <: Real` - and the reason I haven't really advocated it - is that I'd still like to explore using ForwardDiff in complex spaces.

This is a case where either a traits-based approach or multiple inheritance could really save us, because then we could have something like

```julia
abstract ForwardDiffNumber <: Number
abstract ForwardDiffReal <: ForwardDiffNumber, Real # direct subtype of both ForwardDiffNumber and Real

immutable ForwardDiffComplex{F<:ForwardDiffReal} <: ForwardDiffNumber, Complex
    a::F
    b::F
end
```

I mean, I'm not sure that the way I defined `ForwardDiffComplex` makes any sense, but you get the point - the above would completely solve the type hierarchy issue. In fact, we could just do the whole `ForwardDiffReal` thing now, and then use Traits.jl to define `ForwardDiffNumber` if we ever get an implementation of `ForwardDiffComplex` going.

Sorry for the rant - I think this is compelling enough that I'm going to open up an issue in ForwardDiff, and we can continue discussion there.

All that being said, I don't think it should prevent you guys from doing what you're doing here. AFAIK, there's no reason *not* to allow arbitrary number types for `Normal`, especially since you can easily exclude the few known "nonsense" cases with inner constructors (e.g. `Complex`).